src/Tizen.Applications.ComponentBased.Port/Tizen.Applications.ComponentBased.Port/ComponentPort.cs

```diff
@@ -177,34 +249,69 @@ public object SendSync(string endpoint, int timeout, object request)
             {
                 object response = FromParcel(resParcel);
                 return response;
-            }            
+            }
         }
 
         /// <summary>
-        /// Abstract method for receiving a request event.
+        /// Sends the request data and receives the reply data asynchronously.
         /// </summary>
-        /// <param name="sender">The name of the sender</param>
-        /// <param name="request">The serializable data</param>
-        /// <since_tizen> 9 </since_tizen>
-        protected abstract void OnRequestEvent(string sender, object request);
+        /// <exception cref="ArgumentException">Thrown when the argument is invalid.</exception>
+        /// <exception cref="OutOfMemoryException">Thrown when the memory is insufficient.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when because of permission denied.</exception>
+        /// <exception cref="global::System.IO.IOException">Thrown when because of I/O error.</exception>
+        /// <param name="endpoint">The name of the endpoint</param>
+        /// <param name="timeout">The timeout in milliseconds, -1 to use the default timeout</param>
+        /// <param name="request">The serializable data to send</param>
+        /// <returns>The received serializable data</returns>
+        /// /// <since_tizen> 9 </since_tizen>
+        public async Task<object> SendAndReceiveAsync(string endpoint, int timeout, object request)
+        {
+            try
+            {
+                return await Task.Run(() => SendAndReceive(endpoint, timeout, request)).ConfigureAwait(false);
+            }
+            catch
+            {
+                throw;
```

Comment:
what does this try/catch/throw change? for me it looks like it could be dropped. Does it make the callstack in exception better?

Reply:
If there is no try catch, exception throw occurs internally. And then, ends as crash. This is to deliver the exception to the application.